yok.ts

```diff
@@ -137,9 +137,29 @@ export class Yok implements IInjector {
 		forEachName(names, (name) => this.requireOne(name, file));
 	}
 
+	public publicApi: any = {};
+	public _publicApi: any = {};
+
+	public requirePublic(names: any, file: string): void {
+		forEachName(names, (name) => {
+			this.requireOne(name, file);
+			this.resolvePublicApi(name, file);
+		});
+	}
+
+	private resolvePublicApi(name: string, file: string): void {
+		Object.defineProperty(this.publicApi, name, {
+			get: () => {
+				let obj: any = {};
+				let originalModule = this.resolve(name);
```

Comment:
when this.resolve is called, the injector requires the file. During this process, the decorators are executed and in `promisify` decorator we populate _publicApi.